test/loom/test/derived.clj

```diff
@@ -0,0 +1,74 @@
+(ns loom.test.derived
+  (:require [loom.derived :refer :all]
+            [loom.graph :refer :all]
+            [loom.alg :refer :all]
+            [clojure.test :refer :all]))
+
+(deftest derived-graph-test
+  (let [g  (graph [1 2] [1 3] [2 3] 4)
+        dg (digraph [1 2] [1 3] [2 3] 4)]
+    (testing "mapped-by"
+      (are [expected got] (= expected got)
```

Comment:
For each type of function tested, it'd be good to add tests for `false` conditions for both directed and undirected graphs. One check that `false` is returned when appropriate should be sufficient for each type of test.